app/scripts/models/pairing/state.js

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { assign } from 'underscore';
+import { Model } from 'backbone';
+
+/* eslint-disable no-use-before-define */
+export class State extends Model {
+  constructor (attributes, options = {}) {
+    super(attributes, options);
+
+    this.broker = options.broker;
+    this.notifier = options.notifier;
+    this.parent = options.parent;
+    this.oAuthClient = options.oAuthClient;
+    this.relier = options.relier;
+  }
+
+  destroy () {
+    this.stopListening();
+    super.destroy();
+  }
+
+  gotoState (NextState, attrs) {
+    this.trigger('goto.state', NextState, attrs);
+  }
+
+  navigate (url, nextViewData) {
+    this.notifier.trigger('navigate', {
+      nextViewData: assign(this.toJSON(), nextViewData),
+      routerOptions: { replace: true, trigger: true },
+      url,
+    });
+  }
+}
+
+export class CompleteState extends State {
+  name = 'Complete';
+
+  connectionClosed () {
```

Comment:
I don't see `connectionClosed` used anywhere, I do however see [socketClosed](https://github.com/mozilla/fxa-content-server/pull/6927/files#diff-a7e1c12e60434621d29c02a1c4549000R20) in the supplicant-state-machine. Can the method be removed? If so, there is also a reference [in this comment](https://github.com/mozilla/fxa-content-server/pull/6927/files#diff-a7e1c12e60434621d29c02a1c4549000R130)

Reply:
good catch!